### ISSUE TITLE
Unexpected error in specifying kmeansK parameters

### ISSUE THREAD

**Issue by makimenkova** [SydneyBioX/scMerge#13]
Hi,

I'm interested in integrating sc-RNA data sets and your package seems very suitable for this. However I get this error while running scMerge.

`sce_mESC_mg <- scMerge::scMerge(sce_combine = sce_combine,
                                ctl = ctl, 
                                kmeansK = c(1, 2),  
                                assay_name = 'scMerge')`

"Step 1: Computation will run in serial
Error in igraph::fastgreedy.community(g) : 
  At fast_community.c:639 : fast-greedy community finding works only on graphs without multiple edges, Invalid value"

For any other value of the kmeansK parameter everything works, but I need exactly kmeansK = c(1, 2). I tried to change data sets order and specify kmeansK = c(2, 1), but it didn't work

"Step 1: Computation will run in serial
Error in utils::combn(c(seq_len(batch_num))[-batch_oneType], 2) : n < m"

Thanks in advance,
Maria

**Comment by kevinwang09** [comment]
Hi Maria, 
We have some ideas as to why this occurred. However, in order to help us to diagnose the problem, could you run 
`devtools::session_info()`
and post the relevant lines for `scMerge`?
Many thanks, 
Kevin

**Comment by makimenkova** [comment]
Hello! Here is my session_info:
 ```
─ setting  value                       
 version  R version 3.6.1 (2019-07-05)
 os       Ubuntu 18.04.2 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                          

─ Packages 
 scMerge                1.0.0      2019-05-02 [1] Bioconductor 
 ```
Thanks

**Comment by kevinwang09** [comment]
Hi Maria,
That is very helpful, thank you.
Could I ask you to try to install the latest version using `devtools::install_github("SydneyBioX/scMerge, force = TRUE)`?
This should install Version 1.1.3 which might already provided a fix.
Thanks,
Kevin

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by makimenkova** [comment]
Thanks! I did it, but it didn't help unfortunately
``` V scMerge                1.0.0      2019-07-25 [1] Github (SydneyBioX/scMerge@1936764)```
```Step 1: Computation will run in serial
Error in igraph::fastgreedy.community(g) : 
  At fast_community.c:639 : fast-greedy community finding works only on graphs without multiple edges, Invalid value```

**Comment by kevinwang09** [comment]
Hi, 
It seems like you still have scMerge V1.0.0 installed. This is an outdated version, so you should try to run `devtools::install_github("SydneyBioX/scMerge, force = TRUE)`, restart R and try to load scMerge again. You should have V1.1.4 installed after this. 

As for the error itself, I wonder if you see a plot that looks somewhat like this one before the error occur. If so, could you share that with us? 

![image](https://user-images.githubusercontent.com/14134289/62340939-af89bb00-b524-11e9-82f7-1c601c45387d.png)


Thanks, 
Kevin

**Comment by DarioS** [comment]
> ... 1.0.0 ... is an outdated version ...

Not really. Almost all end-users should be using the release branch, which is at 1.0.0. The development branch is for the package's developer and early adopters / testers. Are there changes to the developer branch which ought to be ported to the release branch? Installing using `devtools::install_github` just annoys Martin Morgan on the mailing list when users end up with a mix of incompatible R and R package versions, which wouldn't happen if BiocManager is used.

**Comment by kevinwang09** [comment]
@DarioS, `scMerge V1.0.0` is fine as is. But after the BioC release this year, we went on to expand `scMerge`'s capabilities in V1.1.z. 

Maybe for the sake of consistency, @makimenkova, could you try `BiocManager::install("scMerge", version = "3.10)` instead? Once we resolve this issue, I will push the patched version to BioC.
